install-validator/internal/dir-compare/compare.go

```diff
@@ -0,0 +1,102 @@
+package dir_compare
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/client-go/rest"
+
+	nexuscompare "github.com/vmware-tanzu/graph-framework-for-microservices/common-library/pkg/nexus-compare"
+	ext "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func CheckDir(dir string) (bool, *bytes.Buffer, error) {
+	var changes []*bytes.Buffer
+	incompatible := false
+	cs, err := getKubeClientSet()
+	if err != nil {
+		return true, nil, err
+	}
+	err = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		if !strings.HasSuffix(info.Name(), ".yaml") {
+			return nil
+		}
+		newData, err := os.ReadFile(path)
```

Comment:
Please check/react to error. Maybe we should fail the validation altogether, even if a single file read fails.